### ISSUE TITLE
Minecraft 1.17 Failed to run and crashes

### ISSUE THREAD

**Issue by KozmikNano** [Botspot/pi-apps#738]
Problem: Starting 1.17 in the minecraft launcher results in the crash (log bellow). I had looked at this and attempted to follow those steps but failed. Link to site: [](https://www.raspberrypi.org/forums/viewtopic.php?f=78&t=284448&start=175)
System: `
Twister OS 2.0.2
Raspberry pi 4 8 gb
`
Log: 
```
Java HotSpot(TM) Server VM warning: Using incremental CMS is deprecated and will likely be removed in a future release
Error: A JNI error has occurred, please check your installation and try again
Exception in thread "main" java.lang.UnsupportedClassVersionError: net/minecraft/client/main/Main has been compiled by a more recent version of the Java Runtime (class file version 60.0), this version of the Java Runtime only recognizes class file versions up to 52.0
	at java.lang.ClassLoader.defineClass1(Native Method)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:756)
	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
	at java.net.URLClassLoader.defineClass(URLClassLoader.java:468)
	at java.net.URLClassLoader.access$100(URLClassLoader.java:74)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:369)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:363)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:362)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:418)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:355)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:351)
	at sun.launcher.LauncherHelper.checkAndLoadMain(LauncherHelper.java:495)
Java HotSpot(TM) Server VM warning: Using incremental CMS is deprecated and will likely be removed in a future release
Error: A JNI error has occurred, please check your installation and try again
Exception in thread "main" java.lang.UnsupportedClassVersionError: net/minecraft/client/main/Main has been compiled by a more recent version of the Java Runtime (class file version 60.0), this version of the Java Runtime only recognizes class file versions up to 52.0
	at java.lang.ClassLoader.defineClass1(Native Method)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:756)
	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
	at java.net.URLClassLoader.defineClass(URLClassLoader.java:468)
	at java.net.URLClassLoader.access$100(URLClassLoader.java:74)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:369)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:363)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:362)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:418)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:355)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:351)
	at sun.launcher.LauncherHelper.checkAndLoadMain(LauncherHelper.java:495)
```
Picture of my config:
![image](https://user-images.githubusercontent.com/77017394/122257416-9af30980-ce95-11eb-8690-9c0a74bb83d8.png)

**Comment by chunky-milk** [comment]
I've found a fix to this, it's because Minecraft 1.17 demands OpenJDK 16...
```
# download and unpack java 16 arm32 archive
wget https://github.com/AdoptOpenJDK/openjdk16-binaries/releases/download/jdk16u-2021-05-08-12-45/OpenJDK16U-jdk_arm_linux_hotspot_2021-05-08-12-45.tar.gz
tar -xvf OpenJDK16U-jdk_arm_linux_hotspot_2021-05-08-12-45.tar.gz
rm OpenJDK16U-jdk_arm_linux_hotspot_2021-05-08-12-45.tar.gz

# Open the Minecraft Launcher, click Edit Profile in the bottom left corner
# Change the "Executable" argument to the following, remember to change "pi" to your username if you have a different one
/home/pi/jdk-16.0.1+4/bin/java
# Also, make sure you change the JVM Arguments to the following...
-Dorg.lwjgl.librarypath=/home/pi/lwjgl3arm32 -Xmx1G -XX:-UseAdaptiveSizePolicy -Xmn128M 


# to add it to path use this (not tested)...
cd ~/
export PATH=$PWD/jdk-16.0.1+4/bin:$PATH
```

**Comment by KozmikNano** [comment]
Thanks I will try that out later!

**Comment by KozmikNano** [comment]
Alright it worked out perfectly! @Botspot i would pin this for others to have reference!

**Comment by chunky-milk** [comment]
> Alright it worked out perfectly! @Botspot i would pin this for others to have reference!

Glad I could help 😄

**Comment by Pythonic456** [comment]
Just one small thing... for better performance and (possibly) greater stability, change `-Xmx1G` to `-Xmx2G`, and maybe test `-Xmx4G`.

**Comment by KozmikNano** [comment]
Alright so I could only up the ram by 1gb because I was running a 32bit Java. If I were running 64 bit then I could do 4. I wish I could do more though as I have 8gbs

**Comment by chunky-milk** [comment]
I will send a PR soon with the 1.17 fix

**Comment by Pythonic456** [comment]
@KozmikNano I am using 32-bit RPi OS and I can have 2GB, why doesn't it seem to work for you?

**Comment by KozmikNano** [comment]
Sorry for the confusion I have 2gb I was saying I couldn’t use the 4gb.

**Comment by Pythonic456** [comment]
👍

**Comment by chunky-milk** [comment]
Should be fixed now. Try updating

**Comment by Botspot** [comment]
This issue can be closed because #743 was merged.
